docs/introduction-to-smart-contracts.rst

```diff
@@ -563,7 +563,9 @@ idea, but it is potentially dangerous, as if someone sends Ether to removed
 contracts, the Ether is forever lost.
 
 .. warning::
-    From version 0.8.18 and up, the use of ``selfdestruct`` in both Solidity and Yul will trigger a
+    From ``EVM >= Cancun`` onwards, ``selfdestruct`` will **only** send all Ether in the account to the given recipient. In the case when ``selfdestruct`` is called in the same transaction the contract was created in, the behaviour of ``selfdestruct`` described above the warning pre-Cancun hardfork (i.e., ``EVM <= Shanghai``) is preserved and will destroy the current contract, deleting any data, including storage keys, code and the account itself.
+
+    Also, from version 0.8.18 and up, the use of ``selfdestruct`` in both Solidity and Yul will trigger a
     deprecation warning, since the ``SELFDESTRUCT`` opcode will eventually undergo breaking changes in behavior
     as stated in `EIP-6049 <https://eips.ethereum.org/EIPS/eip-6049>`_.
```

Comment:
Maybe rephrase this to something like:
```suggestion
    Also, note that the ``selfdestruct`` opcode remains deprecated since Solidity version 0.8.18, and the use of the opcode in both Solidity and Yul will trigger a deprecation warning, since the opcode is expected to undergo substantial behavioral changes, as stated in `EIP-6049 <https://eips.ethereum.org/EIPS/eip-6049>`_.
    Moreover, future adjustments to the EVM may further restrict the functionality of this opcode.
```